Delegate #readbyte on the abstract IO adapter, and improve documentation around custom processors

creature opened the pull request:
I recently implemented a custom processor that used [the Exifr gem](https://github.com/remvee/exifr) to extract EXIF information from images uploaded to a Paperclip attachment. Exifr's processor uses `readbyte` to parse the EXIF header, so it hit errors when Paperclip's `File`-like object didn't have one. There's also a test for this delegation. 

I've also tidied up the README documentation for custom processors, hopefully to be more clear than before. There was some duplicated content between the "Post Processing" section and the "Custom Attachment Processing" section, and those sections were separated in the file. I've dedicated the "Post Processing" section to Paperclip's built in thumbnailing processors, and made "Custom Attachment Processing" section solely about writing & using your own custom processors. This should be a more understandable progression, as built-in functionality is discussed first & separately from extending Paperclip.

I've also tidied up a couple of inaccuracies in the `Paperclip::Processor` documentation, and generally applied Strunk & White's "Omit needless words" principle in the areas I've touched.

creature commented:
Are there any changes or amendments you'd like to see to this? I'm happy to tweak it as required to get this merged.

tute commented:
Thank you, @creature! This is great. Can you please rebase on top of latest master, and squash your commits together? Also, if you can, please add your good PR description as commit message. (Check https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history if needed).

Can you add context on the `:readbyte` addition? Thanks, again! 😃

creature commented:
Hi! Sorry for the slow reply. I'll try to rebase & squash the commits as you requested some time soon. 

As for context on `readbyte`, it's a [method defined on Ruby's `IO` class](http://ruby-doc.org/core-2.2.0/IO.html#method-i-readbyte), and `File < IO`, so that's why Exifr presumes it will be available. Is that what you had in mind, or am I off-base? :)

creature commented:
@tute I've rebased this onto the master branch, squashed it down to a single commit, and included a slightly-abridged form of my PR comment as the commit message. Let me know if there's any further changes you'd like to see to this! :)

tute commented:
Thank you!
